src/ControllerV2/RosalieInteroperabilityController.php

```diff
@@ -63,6 +63,18 @@ private function processSiSiaoNumberForm(Request $request, Beneficiaire $benefic
         ]);
     }
 
+    private function processBeneficiaryCheckResult(BeneficiaryCheckOnRosalie $beneficiaryCheck): void
+    {
+        if ($beneficiaryCheck->beneficiaryIsFound()) {
+            $this->service->linkBeneficiaryToRosalie($beneficiaryCheck->getBeneficiary());
+            $this->addFlash('success', 'si_siao_number_found_rosalie');
+        } else {
+            $this->addFlash('error', $beneficiaryCheck->getErrorMessage());
+        }
+
+        $this->em->flush();
```

Comment:
Why is there a flush here ?

Reply:
To flush beneficiary hydration with SISIAO key

Reply:
I mean why not put it in the service ?

Reply:
Oh no its clear now that it's not in a separated method. My bad